.github/workflows/ccpp.yml

```diff
@@ -1,28 +1,311 @@
 name: C/C++ CI
 
-on: [push]
+on: [push, pull_request]
 
 jobs:
-  build:
+  build-ubuntu-default:
```

Comment:
`default` is supposed to be about checking pure lib source distribution with plain configure & make. So can we remove '-fPIC' and swig here?

also, adding simple description/comment for each job can be useful for future maintenance, e.g:
```
jobs:
  build-ubuntu-default:
  # checking pure lib source distribution with plain configure & make
  ...
```